### PR TITLE
(#32) Ensure null weak pointers are removed from a Node parent list

### PR DESCRIPTION
**Pull request by bluemarvin**

### CODE REVIEW

File: src/Node.cpp [hunk 1]
```diff
@@ -53,15 +54,16 @@ Node::AddToParents(GroupWeak& aParent, Node& aChild) {
...
-  GroupPtr parent = aParent.lock();
-  if (!parent) {
-    return;
-  }
-  for (auto it = aChild.m.parents.begin(); it != aChild.m.parents.end(); it++) {
-    if (it->lock() == parent) {
-      aChild.m.parents.erase(it);
+Node::RemoveFromParents(Group& aParent, Node& aChild) {
+  for (auto it = aChild.m.parents.begin(); it != aChild.m.parents.end();) {
+    Group* node = it->lock().get();
```
Comment (MortimerGoro) [inline src/Node.cpp]: It would be safer to create a extra variable with the lock result, and the do the get().

Comment (bluemarvin) [reply src/Node.cpp]: I thought as long as it != end(), *it will always return a valid object? Is that not the case?

Comment (MortimerGoro) [reply src/Node.cpp]: My concern was about accessing `Group* node` contents without a shared_str protecting the value (the one returned from lock() get´s destroyed). But if we are only using the value for comparations and not actually using it, it's ok.

**Comment by MortimerGoro** [comment]
There is some conflict in Node.cpp

**Comment by bluemarvin** [comment]
Rebased.

**Pull request merged**